### ISSUE TITLE
Not a very good name for the option to choose between jp2 and j2k in JPEG2000 export

### ISSUE THREAD

**Issue by victoryforce** [darktable-org/darktable#14647]
As you can see in this screenshot, we have two drop-down menus with very similar widget labels:

<img width="432" alt="darktable_fHxjX7SE15" src="https://github.com/darktable-org/darktable/assets/6246352/5af38b42-cc1e-49ce-9ff2-684e651483bf">

It is very difficult to understand why we have two "formats", why one is a "file format" and the other is just a format. And if the choice between jp2 and j2k in the second menu is not a choice between different file formats, then what is "just format" then?

In fact, defining any image format involves defining two things:
- How, with what algorithms, images are coded and decoded
- How this data is stored in the file. For example, we have to define the magic bytes, how to define the start offset of the actual image code in the file, what metadata can be, what their format is.

In short, an image format is a standard that precisely specifies the encoder/decoder and the container.

In this particular case, jp2 and j2k are different container options. More specifically, jp2 is the standard file format (container) for JPEG2000 images, while j2k stands for JPEG2000 codestream.

Thus, it is better to rename "file format" in the export module to simply "format", and what is called "format" in export to JPEG2000 should be renamed to "container" (container and file format are essentially synonymous here, but it is better to choose a term without the word "format ", so as not to leave similarities in names and not to confuse the user).

**Comment by MStraeten** [comment]
why not simply ‚file extension‘ - container is a technical term.

**Comment by victoryforce** [comment]
> why not simply ‚file extension‘ - container is a technical term.

Because it's not just a file extension. You can export both versions and compare the contents of the files, it will be different. jp2 will contain the xmp metadata block, j2k will not. The magic bytes will indicate that one of the files contains a JPEG 2000 Part 1 (Core) file and the other a JPEG 2000 Codestream.

I agree that "container" is a more technical term than "file format", but the fact is that if the user deliberately chooses a non-default option in this widget, then he knows what he is doing and knows the differences of j2k, and this is quite technical knowledge.

I reported an issue specifically about the suboptimal choice of names for a couple of widgets, but another issue can be discussed, whether there is really a need to export to j2k. I have no idea what was the purpose of implementing the option to export the image specifically to the codestream version of the JPEG2000 files. Why might a user need such a version?

**Comment by kmilos** [comment]
This should probably at least be changed to a yes/no dropdown (or a checkbox) saying something like "naked codestream". The resulting "jp2/j2k" extension should probably be hidden from the user IMHO...

> the fact is that if the user deliberately chooses a non-default option in this widget, then he knows what he is doing and knows the differences of j2k, and this is quite technical knowledge

Correct.

> I reported an issue specifically about the suboptimal choice of names for a couple of widgets, but another issue can be discussed, whether there is really a need to export to j2k. I have no idea what was the purpose of implementing the option to export the image specifically to the codestream version of the JPEG2000 files. Why might a user need such a version?

No idea either, especially since it doesn't support metadata... In the newer JPEG XL export code, we didn't even consider the naked codestream option IIRC...
